src/local/test.py

```diff
@@ -0,0 +1,59 @@
+import hydra
+from omegaconf import OmegaConf
+import mlflow
+import numpy as np
+from pytorch_lightning import Trainer
+import torch
+from sklearn.metrics import r2_score
+
+from local.experiment import LassoNetExperiment
+
+@hydra.main(config_path='configs', config_name='test')
+def test(cfg: DictConfig):
+    '''
+    Load a lassonet model from the specified node of a train experiment, then test it on a different node
+    and log the results.
+    '''
+    print(cfg)
+    
+    experiment = LassoNetExperiment(cfg)
+    experiment.load_sample_indices()
+    experiment.load_data()
+    
+    print("Loading train experiment")
+    train_experiment = mlflow.get_experiment_by_name(cfg.experiment.train_name)
+    runs = mlflow.list_run_infos(train_experiment.experiment_id)
+    df = mlflow.search_runs(experiment_ids=[train_experiment.experiment_id])
+
+    # Check that the selected training run exists and is unique
+    assert(sum(df['tags.node_index'] == str(cfg.train_node_index)) == 1)
+    mlflow_run_id = df.loc[(df['tags.node_index'] == str(cfg.train_node_index)) and (df['tags.snp_count'] == str(cfg.experiment.snp_count)), 'run_id'].values[0]
+    run = mlflow.get_run(mlflow_run_id)
+    loaded = mlflow.pytorch.load_model(f"runs:/{mlflow_run_id}/lassonet-model")
+    loaded.eval()
+    
+    print("Predicting")
+    
+    trainer = Trainer(gpus=0)
+    train_preds, val_preds, test_preds = trainer.predict(loaded, experiment.data_module)
+
+    best_col = int(run.data.metrics['best_alpha_index'])
+    best_val_r2 = r2_score(experiment.y_val, torch.cat(val_preds)[:, best_col].numpy())
```

Comment:
in the future we will need the metric function and name here and below to be parameters and not hardcoded. you can fix it now or when you deal with non-continuous phenotypes, up to you. I think in other parts of the code I have function arguments of metric_fun=r2_score and metric_name='r2'. This is what I used for TG (categorical output)